src/fcc_bridge_node_safety.cpp

```diff
@@ -261,61 +296,34 @@ void FCCBridgeNode::check_gps_state() {
             }
             break;
         default:
-            throw std::runtime_error(
+            throw unknown_enum_value_error(
                 std::string("Got invalid value for internal_state: ") +
                 std::to_string(static_cast<int>(this->get_internal_state())));
     }
 
-    switch (this->get_internal_state()) {
-        case INTERNAL_STATE::ERROR:
-            // This should never happen, as the process exits on ERROR state
-            throw std::runtime_error(std::string(__func__) +
-                                     " called while in ERROR state");
-            // This function should never be called in these states
-        case INTERNAL_STATE::STARTING_UP:
-        case INTERNAL_STATE::ROS_SET_UP:
-            RCLCPP_FATAL(this->get_internal_state_logger(),
-                         "In an invalid state to check GPS! Exiting...");
-            this->set_internal_state(INTERNAL_STATE::ERROR);
-            this->exit_process_on_error();
-        case INTERNAL_STATE::MAVSDK_SET_UP:
-            break;
-        case INTERNAL_STATE::ARMED:
-        case INTERNAL_STATE::WAITING_FOR_ARM:
-        case INTERNAL_STATE::LANDED:
-            // Check that the current coordinate is inside the geofence.
-            if (!this->check_point_in_geofence(
-                    this->last_fcc_position.value())) {
-                // The current point is outside the geofence and the UAV is not
-                // airborne
-                RCLCPP_FATAL(this->get_safety_logger(),
-                             "The current position is not inside the geofence! "
-                             "UAV is not airborne. Exiting...");
-                this->set_internal_state(INTERNAL_STATE::ERROR);
-                this->exit_process_on_error();
-            }
-            break;
-        case INTERNAL_STATE::TAKING_OFF:
-        case INTERNAL_STATE::WAITING_FOR_COMMAND:
-        case INTERNAL_STATE::FLYING_MISSION:
-        case INTERNAL_STATE::LANDING:
-        case INTERNAL_STATE::RETURN_TO_HOME:
-            // Check that the current coordinate is inside the geofence.
-            if (!this->check_point_in_geofence(
-                    this->last_fcc_position.value())) {
+    // In state MAVSDK set up no safety limits are configured so a geofence
```

Comment:
I think we should add an else case for this state in order to put the situation into the log accordingly. Currently it will say "GPS state is O.K" in that state no matter what the circumstances are.

Reply:
I added a log statement in 9f292b8
The "GPS state is O.K" message would only be printed in the case everything is ok.
If the point is outside the geofence both cases of the airborne check would prematurely end the execution of the check_gps_state function.

Reply:
Thanks. But don't we go from line 307 (the if statement) directly to line 333, the "    RCLCPP_INFO(this->get_safety_logger(), "GPS state is O.K.");" log, in case this->get_internal_state() evaluates to INTERNAL_STATE::MAVSDK_SET_UP? Therefore in case we are in this state, we always should get GPS state is O.K. in the log. Is that correct or am I getting something wrong here?

Reply:
I am a bit confused. The MAVSDK_SET_UP case is handled in Line 281. There is no if statement in Line 307 in both the old and new version of the file and Line 333 is past the end of the current function. At least in the view I am getting.

Reply:
Fixed in 2b9d184